### PR TITLE
Add DataQueryable and DataIndexed states to bulk load tasks

### PR DESCRIPTION
**Pull request by soothing-rain**
issue: #16607

/kind enhancement

Signed-off-by: Yuchen Gao <yuchen.gao@zilliz.com>

### CODE REVIEW

**Comment by soothing-rain** [comment]
rerun ut

File: internal/rootcoord/root_coord.go [hunk 1]
```diff
@@ -372,15 +369,15 @@ func (c *Core) checkFlushedSegments(ctx context.Context) {
 			ctx2, cancel2 := context.WithTimeout(ctx, 3*time.Minute)
 			segIDs, err := c.CallGetFlushedSegmentsService(ctx2, collMeta.ID, partID)
 			if err != nil {
-				log.Debug("failed to get flushed segments from data coord",
```
Comment (congqixia) [inline internal/rootcoord/root_coord.go:375]: NIT: log level shall be WARN at least.

File: internal/rootcoord/root_coord.go [hunk 2]
```diff
@@ -2489,61 +2487,86 @@ func (c *Core) CountCompleteIndex(ctx context.Context, collectionName string, co
...
 		case <-ticker.C:
-			log.Info("(in check complete index loop) check segments' index states", zap.Int64("task ID", ti.GetId()))
-			if ct, err := c.CountCompleteIndex(ctx, colName, ti.GetCollectionId(), segIDs); err == nil &&
-				segmentsOnlineReady(ct, len(segIDs)) {
-				log.Info("segment indices are ready",
-					zap.Int64("task ID", ti.GetId()),
-					zap.Int("total # of segments", len(segIDs)),
-					zap.Int("# of segments with index ready", ct))
-				c.bringSegmentsOnline(ctx, segIDs)
+			log.Info("(in check segment loaded loop) check segments' loading states",
```
Comment (congqixia) [inline internal/rootcoord/root_coord.go:2517]: NIT: maybe to frequent?

File: internal/rootcoord/root_coord.go [hunk 3]
```diff
@@ -2489,61 +2487,86 @@ func (c *Core) CountCompleteIndex(ctx context.Context, collectionName string, co
...
+	defer ticker.Stop()
+	expireTicker := time.NewTicker(time.Duration(Params.RootCoordCfg.ImportIndexWaitLimit*1000) * time.Millisecond)
+	defer expireTicker.Stop()
+	for {
+		select {
+		case <-c.ctx.Done():
+			log.Info("(in check complete index loop) context done, exiting checkCompleteIndexLoop")
+			return
+		case <-ticker.C:
+			log.Info("(in check complete index loop) check segments' index states",
```
Comment (congqixia) [inline internal/rootcoord/root_coord.go:2555]: Maybe too frequent?

**Review by congqixia — APPROVED** [review]
/lgtm

File: internal/rootcoord/root_coord.go (hunk 1 above)
Comment (soothing-rain) [reply internal/rootcoord/root_coord.go:375]: Sure, will address.

**Pull request merged**